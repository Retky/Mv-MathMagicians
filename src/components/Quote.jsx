import React from 'react';
import './quote.css';

const Quote = () => (
  <p className="quote">
    Mathematics is not about numbers, equations, computations, or algorithms; it is about
    understanding.
    <br />
    <span> -William Paul Thurston</span>
  </p>
);

export default Quote;
