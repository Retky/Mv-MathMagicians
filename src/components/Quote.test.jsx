import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Quote from './Quote';

describe('Home component', () => {
  it('renders "Mathematics is not about numbers..."', () => {
    // Arrange
    render(<Quote />);

    // Assert
    const headingElement = screen.getByText('Mathematics is not about numbers', { exact: false });
    expect(headingElement).toBeInTheDocument();
  });

  it('renders component to match the previous snapshot', () => {
    // Assert
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
