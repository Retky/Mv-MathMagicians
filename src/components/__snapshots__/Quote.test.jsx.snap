// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders component to match the previous snapshot 1`] = `
<p
  className="quote"
>
  Mathematics is not about numbers, equations, computations, or algorithms; it is about understanding.
  <br />
  <span>
     -William Paul Thurston
  </span>
</p>
`;
